crates/red_knot_python_semantic/src/types/display.rs

```diff
@@ -292,7 +292,14 @@ impl Display for DisplayUnionType<'_> {
                     db: self.db,
                 });
             } else {
-                join.entry(&element.display(self.db));
+                if let Type::Callable(
+                    CallableType::General(_) | CallableType::MethodWrapperDunderGet(_),
+                ) = element
+                {
+                    join.entry(&format!("({})", element.display(self.db)));
```

Comment:
Ah, I wish [`from_fn`](https://doc.rust-lang.org/std/fmt/fn.from_fn.html) gets stabilizied. It would allow to remove the `format!` without much ceremony.... 

Although, I think you can avoid the `format` call here by using `format_args` (but not a 100% sure)
```suggestion
                    join.entry(&format_args!("({})", element.display(self.db)));
```